src/commands/synthetics/mobile.ts

```diff
@@ -1,7 +1,88 @@
 import * as crypto from 'crypto'
+import * as fs from 'fs'
+
+import {APIHelper} from './api'
+import {Test, TestPayload} from './interfaces'
+import {getTestByPublicId} from './utils'
 
 export const getMD5HashFromFileBuffer = async (fileBuffer: Buffer): Promise<string> => {
   const hash = crypto.createHash('md5').update(fileBuffer).digest('base64')
 
   return hash
 }
+
+export const uploadMobileApplications = async (
+  api: APIHelper,
+  applicationPathToUpload: string,
+  mobileApplicationId: string
+): Promise<string> => {
+  const fileBuffer = await fs.promises.readFile(applicationPathToUpload)
+  const md5 = await getMD5HashFromFileBuffer(fileBuffer)
+  const {presigned_url_params: presignedUrl, file_name: fileName} = await api.getMobileApplicationPresignedURL(
+    mobileApplicationId,
+    md5
+  )
+
+  await api.uploadMobileApplication(fileBuffer, presignedUrl)
+
+  return fileName
+}
+
+export const uploadApplicationIfNeeded = async (
+  api: APIHelper,
+  applicationPathToUpload: string,
+  test: Test,
+  uploadedApplicationByApplication: {[applicationFilePath: string]: {applicationId: string; fileName: string}[]}
+) => {
+  const isAlreadyUploaded =
+    applicationPathToUpload in uploadedApplicationByApplication &&
+    uploadedApplicationByApplication[applicationPathToUpload].find(
+      ({applicationId}) => applicationId === test.options.mobileApplication!.applicationId
+    )
+
+  if (!isAlreadyUploaded) {
```

Comment:
As a nit, you can return early in case upload is not needed to spare one indentation level